sw/host/opentitanlib/src/transport/hyperdebug/i2c.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{ensure, Result};
+use std::rc::Rc;
+use zerocopy::{AsBytes, FromBytes};
+
+use crate::io::i2c::{I2cError, Bus, Transfer};
+use crate::transport::hyperdebug::{BulkInterface, Error, Hyperdebug, Inner};
+
+pub struct HyperdebugI2cTarget {
+    inner: Rc<Inner>,
+    interface: BulkInterface,
+    bus_idx: u8,
+    max_write_size: usize,
+    max_read_size: usize,
+}
+
+const USB_MAX_SIZE: usize = 64;
+
+/// Wire format of USB packet to request a short I2C transaction
+/// (receiving at most 127 bytes).
+#[derive(AsBytes, FromBytes, Debug)]
+#[repr(C)]
+struct CmdTransferShort {
+    port: u8,
+    addr: u8,
+    write_count: u8,
+    read_count: u8,
+    data: [u8; USB_MAX_SIZE - 4],
+}
+impl CmdTransferShort {
+    fn new() -> Self {
```

Comment:
This can just be done with `derive(Default)`.

Reply:
If I try that, I get an error that "the trait `Default` is not implemented for `[u8; 60]`" at the data member.

Reply:
Ah right this. Sigh. Hopefully we'll fix that soon.